forest/src/main.rs

```diff
@@ -6,23 +6,118 @@ mod daemon;
 mod logger;
 mod subcommand;
 
-use cli::{cli_error_and_die, Cli};
+use cli::{cli_error_and_die, Cli, DaemonConfig};
+
+use async_std::task;
+use daemonize_me::{Daemon, DaemonError, Group, User};
+use log::info;
+use raw_sync::events::{Event, EventInit};
+use raw_sync::Timeout;
+use shared_memory::ShmemConf;
 use structopt::StructOpt;
 
-#[async_std::main]
-async fn main() {
+use std::fs::File;
+use std::process;
+use std::time::Duration;
+
+const EVENT_TIMEOUT: Timeout = Timeout::Val(Duration::from_secs(20));
+
+// The parent process and the daemonized child communicate through an Event in
+// shared memory. The identity of the shared memory object is written to a local
+// file named .forest_daemon_ipc. The parent process is responsible for cleaning
```

Comment:
So by default, it will create this file in the CWD? If so, I'd say we should change it to some temp dir. E.g. something like this would fail, right? `cd / && forest --detach`. Unless it's a common practice do it like this.

Reply:
Yes it was CWD. Using `temp_dir()` now.